core/device_hooks/common_kernels.inc.cpp

```diff
@@ -213,6 +214,28 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_CG_STEP_2_KERNEL);
 }  // namespace cg
 
 
+namespace bicg {
+
+
+template <typename ValueType>
+GKO_DECLARE_BICG_INITIALIZE_KERNEL(ValueType)
+GKO_NOT_COMPILED(GKO_HOOK_MODULE);
+GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_BICG_INITIALIZE_KERNEL);
+
+template <typename ValueType>
+GKO_DECLARE_BICG_STEP_1_KERNEL(ValueType)
+GKO_NOT_COMPILED(GKO_HOOK_MODULE);
+GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_BICG_STEP_1_KERNEL);
+
+template <typename ValueType>
+GKO_DECLARE_BICG_STEP_2_KERNEL(ValueType)
+GKO_NOT_COMPILED(GKO_HOOK_MODULE);
+GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_BICG_STEP_2_KERNEL);
```

Comment:
There is something weird, you template the BiCG class by `IndexType`, so technically these should also have `IndexType` here? Another possibility, you don't really need the `IndexType` and could remove it.